scheduler/scheduler.go

```diff
@@ -41,80 +42,109 @@ type (
 		PopulateHookData(data map[string]interface{})
 	}
 
+	SchedulerType string
+
 	// Scheduler common core of schedulers
 	Scheduler struct {
 		// SchedType type of scheduler
-		SchedType Type `json:"type" doc-key:"config.scheduler.type"`
+		SchedType SchedulerType `json:"type" doc-key:"config.scheduler.type"`
 		// TimeBuf add wait time in between iterations
 		TimeBuf TimeBuffer `json:"iterationtimebuffer" doc-key:"config.scheduler.iterationtimebuffer"`
 		// InstanceNumber used to ensure different randomizations when running script in multiple different instances
 		InstanceNumber uint64 `json:"instance" doc-key:"config.scheduler.instance"`
 		// ReconnectSettings settings for re-connecting websocket on unexpected disconnect
 		ReconnectSettings session.ReconnectSettings `json:"reconnectsettings" doc-key:"config.scheduler.reconnectsettings"`
 
-		connectionSettings *connection.ConnectionSettings
-		continueOnErrors   bool
+		ConnectionSettings *connection.ConnectionSettings `json:"-"`
```

Comment:
I guess we could add a struct tag for ignoring if this needs to be public.

Reply:
hmm strange json:"-" already means ignore for json

Reply:
It doesn't go through json

Reply:
Although I guess maybe we could add something to look for this struct tag and ignore based on that